labcodes/confetti-ds
# Add support for node v18

**Link to task:** [Card #455 from Jira](https://labcodes.atlassian.net/browse/CSYS-455)

**Staging app:** https://feature-node-18--confetti-storybook.netlify.app/

**How to test:** 
- checkout to branch
- rerun `nvm use` (you may need to `nvm install` too)
- delete all `node_modules` folders
- from the root folder, run `npm i`
- from the root folder, run `npx turbo start`
- verify that the storybook is running correctly
- from the root folder, run `npx turbo test`
- check that all tests pass

**Description of your solution:**

There were a couple of things that were preventing us from using node v18 (and pnpm, if you use it):
- there were illegal imports to libs that weren't used anymore (`@babel` stuff)
- there were a few dependencies that were only working because of v16's dependency tree algorithm (some css and style loaders)
- because we still use webpack v4 on storybook, it also needed an env var to use the correct crypto algorithm (`NODE_OPTIONS=--openssl-legacy-provider`).

So these were all fixed.

For pnpm people (hi 👋 ), I also added workspace configs and a lock file. Don't worry about maintaining it - I'll do that myself, if it ever needs to change :]
